classes/patterns/class-block-pattern.php

```diff
@@ -27,23 +27,37 @@ abstract public static function get_name(): string;
 	abstract public static function get_config( $params = [] ): array;
 
 	/**
-	 * Pattern constructor.
+	 * Returns the pattern classname used for tracking patterns.
 	 */
-	public static function register_all() {
-		if ( ! function_exists( 'register_block_pattern' ) ) {
-			return;
-		}
+	public static function get_classname(): string {
+		return 'is_pattern-' . preg_replace( '#[^_a-zA-Z0-9-]#', '_', static::get_name() );
```

Comment:
Thinking of CSS class names WP conventions, what if we use `-` instead of `_`? For instance the `is-style-*` WP's rule.

Reply:
Oh sorry I misunserstood and misread, you were talking about is_pattern -> is-pattern or about the char replacement '_' ?

Reply:
I meant to the entire `string` because of a CSS `classnames`'s conventions and using `_` it's more related to BEM. What do you think?

Reply:
Yep good point, works for me :+1: